socketio/src/client/client.rs

```diff
@@ -585,143 +735,115 @@ mod test {
     fn socket_io_auth_builder_integration() -> Result<()> {
         let url = crate::test::socket_io_auth_server();
         let nsp = String::from("/admin");
-        let socket = ClientBuilder::new(url)
+        let (tx, rx) = mpsc::sync_channel(1);
+        let _socket = ClientBuilder::new(url)
             .namespace(nsp.clone())
             .auth(json!({ "password": "123" }))
+            .on_any(move |event, payload, _client| {
+                tx.send((String::from(event), payload)).unwrap();
+            })
             .connect_manual()?;
 
-        let mut iter = socket
-            .iter()
-            .map(|packet| packet.unwrap())
-            .filter(|packet| packet.packet_type != PacketId::Connect);
-
-        let packet: Option<Packet> = iter.next();
-        assert!(packet.is_some());
-
-        let packet = packet.unwrap();
-
-        assert_eq!(
-            packet,
-            Packet::new(
-                PacketId::Event,
-                nsp.clone(),
-                Some("[\"auth\",\"success\"]".to_owned()),
-                None,
-                0,
-                None
-            )
-        );
```

Comment:
suggestion: The iterator SHOULD be tested by the unit tests, the message pipe is clever, but more complicated that needed for these tests. 

It's two separate concerns:
- testing behavior (via the iterator)
- callback interface is functioning 

If the callback code breaks, it's much more difficult to debug than the raw iterator implementation.